snuba/datasets/storages/processors/replaced_groups.py

```diff
@@ -128,8 +131,13 @@ def process_query(self, query: Query, request_settings: RequestSettings) -> None
                 logger.warning(
                     "Groups discrepancy between project_extension and processor.",
                     extra={
-                        "extension": existing_groups_conditions,
-                        "processor": condition_to_add,
+                        "extension": deepcopy(existing_groups_conditions),
+                        "processor": deepcopy(condition_to_add),
```

Comment:
I am observing cases where this message gets logged and the two fields are clearly equal.
The only reason I can think for this is that `existing_groups_conditions` (which is a reference to something in the query) changes between this statement and the time the event payload is serialized.